### ISSUE TITLE
Option-marker fallback builds an untrained scoring head silently when option_marker.pt can't be fetched

### ISSUE THREAD

**Issue by MrJev** [wfzyx/von#8]
Follow-up to #5. I re-read the safetensors header on the Hub at `94f2207` and the restore holds — 174 tensors, 395,834,371 parameters, `classifier.weight [3, 1024]`, `classifier.bias`, `head.dense.weight` and `head.norm.weight` all present, so the default path loads with nothing left to initialise. Thank you for turning that around in an afternoon. (I also had the `uv sync --dev` / `pytest` mismatch on my list, and saw you fixed it in `13fbfc4` before I got to it.)

One thing from the same area is still open, and it is the shape of #5 rather than its cause: **when the option-marker weights cannot be fetched, the backend builds the model anyway with an untrained scoring head, and says nothing.**

`src/von/backends/option_marker_backend.py:59-78`:

```python
else:
    # Download from Hugging Face Hub
    try:
        from huggingface_hub import hf_hub_download
        cached_pt = hf_hub_download(repo_id="wfzyx/von-1.0", filename="option_marker.pt")
        model = OptionMarkerModel(base_model_id="wfzyx/von-1.0")
        state_dict = torch.load(cached_pt, map_location=self.device, weights_only=True)
        model.load_state_dict(state_dict, strict=False)
    except Exception:
        base_id = "checkpoints/von-modernbert-rlcd" if os.path.exists("checkpoints/von-modernbert-rlcd") else "wfzyx/von-1.0"
        model = OptionMarkerModel(base_model_id=base_id)
```

`OptionMarkerModel.__init__` constructs `OptionMarkerScorer`, which is `Linear → GELU → LayerNorm → Dropout → Linear`, freshly initialised. In the `except` branch nothing is loaded into it, nothing is logged, and `von serve` then answers normally through the default backend.

The encoder still has to come from somewhere, so this needs the encoder to be reachable while `option_marker.pt` is not — a warm HF cache with `HF_HUB_OFFLINE=1`, a local `checkpoints/von-modernbert-rlcd`, a 401/403 or rate-limit on the single file, or a download interrupted after the encoder and before the `.pt`. That is not exotic for anything long-running.

What that costs, with the scorer built exactly as the fallback builds it (three fixed option representations, six fresh instantiations, CPU, no downloads):

```
fresh scorer 1: logits = [-0.1031, 0.4184, 0.3816] -> option 1
fresh scorer 2: logits = [ 0.5124, -0.0345, 0.0425] -> option 0
fresh scorer 3: logits = [ 0.1358, -0.1072, -0.2974] -> option 0
winning option across 6 fresh instantiations: [1, 0, 0, 2, 0, 0]
```

Same input, different answer per process. From the caller's side it looks like a working von-option-marker server.

**`strict=False` widens the same hole.** Both load paths (`:67` and `:75`) pass it, so a `.pt` that is missing the `scorer.*` keys — a stale checkpoint, a partial write, an encoder-only export like the one behind #5 — loads quietly with a random head instead of raising. That is exactly the failure mode #5 had, and `strict=True` would have caught it at load time.

Two suggestions, either of which closes it:

1. Let the failure be a failure: re-raise with a message naming `option_marker.pt`, or fall back explicitly to the `modernbert` backend with a warning, since that one's weights are complete and it is an honest downgrade rather than a silent one.
2. Load with `strict=True`, and print one line saying which path supplied the weights (local checkpoint, Hub, or none) at first load.

Happy to send a PR for either shape — say which you prefer.

Checked at `14d0987`; no Jev or TypeSafe calls, everything local.

**Comment by wfzyx** [comment]
Fixed in commit `d3874e3`.

### Changes made
1. **No silent untrained fallback:** Removed the silent `except` branch that fell back to instantiating an unweighted `OptionMarkerScorer`. If neither local `option_marker.pt` nor the Hugging Face Hub download succeeds, it now fails fast with an explicit `RuntimeError` refusing to serve with a random scoring head.
2. **`strict=True` enforcement:** Both local and Hub loading paths now use `strict=True` so missing or mismatched weight keys (like the missing scorer head in #5) cannot pass silently.
3. **Weight provenance logging:** Added explicit startup logging stating the exact weight source (`local file` or `Hugging Face Hub`) upon first model initialization.
